crates/iota-names/src/config.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use iota_types::{
+    TypeTag,
+    base_types::{IotaAddress, ObjectID},
+    supported_protocol_versions::Chain,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::Domain;
+
+pub const LEAF_EXPIRATION_TIMESTAMP: u64 = 0;
+pub const DEFAULT_TLD: &str = "iota";
+pub const ACCEPTED_SEPARATORS: [char; 2] = ['.', '*'];
+pub const IOTA_NEW_FORMAT_SEPARATOR: char = '@';
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub struct IotaNamesConfig {
+    pub package_address: IotaAddress,
+    pub object_id: ObjectID,
+    pub payment_package_address: IotaAddress,
+    pub registry_id: ObjectID,
+    pub reverse_registry_id: ObjectID,
+    pub subdomain_proxy_package_id: ObjectID,
+}
+
+impl Default for IotaNamesConfig {
+    fn default() -> Self {
+        // TODO change to mainnet
+        Self::devnet()
+    }
+}
+
+impl IotaNamesConfig {
+    pub fn new(
+        package_address: IotaAddress,
+        object_id: ObjectID,
+        payment_package_address: IotaAddress,
+        registry_id: ObjectID,
+        reverse_registry_id: ObjectID,
+        subdomain_proxy_package_id: ObjectID,
+    ) -> Self {
+        Self {
+            package_address,
+            object_id,
+            payment_package_address,
+            registry_id,
+            reverse_registry_id,
+            subdomain_proxy_package_id,
+        }
+    }
+
+    pub fn from_chain(chain: &Chain) -> Self {
+        match chain {
+            // Chain::Mainnet => IotaNamesConfig::mainnet(),
+            // Chain::Testnet => IotaNamesConfig::testnet(),
+            Chain::Unknown => IotaNamesConfig::devnet(),
+            _ => todo!("uncomment Mainnet and Testnet when IOTA-Names is published there"),
+        }
+    }
+
+    pub fn record_field_id(&self, domain: &Domain) -> ObjectID {
+        let domain_type_tag = Domain::type_(self.package_address);
+        let domain_bytes = bcs::to_bytes(domain).unwrap();
+
+        iota_types::dynamic_field::derive_dynamic_field_id(
+            self.registry_id,
+            &TypeTag::Struct(Box::new(domain_type_tag)),
+            &domain_bytes,
+        )
+        .unwrap()
+    }
+
+    pub fn reverse_record_field_id(&self, address: &IotaAddress) -> ObjectID {
+        iota_types::dynamic_field::derive_dynamic_field_id(
+            self.reverse_registry_id,
+            &TypeTag::Address,
+            address.as_ref(),
+        )
+        .unwrap()
+    }
+
+    // TODO add mainnet() and testnet()
```

Comment:
add the tracking issue.

Reply:
Done in https://github.com/iotaledger/iota/pull/6530